load test.case

class TestRaise : Test.Case {
	const def testNullException(self) {
		var i = null
		while i is null or i < 5 {
			try {
				i++
			} catch e {
				e = 0
			}
		}
		self.expectEqual(5, i)
	}

	const def testMultiException(self) {
		for i in 0..5 {
			a = null
			try {
				a++
			} catch e {
				e = i
			}
			self.expectEqual(i, a)
		}
	}

	const def testRaiseResetScopedSymbols(self) {
		try {
			if let var foo = true {
				raise foo
			}
		} catch e {
			self.expectEqual(true, e)
		}
		self.expectNotDefined(foo)
	}

	const def testRaiseWithoutCatch(self) {
		var test = 7357
		try {
			raise test
		}
		self.expectEqual(7357, test)
	}
}